subsys/net/openthread/Kconfig

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+
+if NET_L2_OPENTHREAD
+
+menu "OpenThread"
+
+config OPENTHREAD_NRF_SECURITY
+	bool "Enable nrf_security module for OpenThread"
+	default y
+	depends on !OPENTHREAD_MBEDTLS
+	select NORDIC_SECURITY_BACKEND
+	select OBERON_BACKEND
+	select MBEDTLS_VANILLA_BACKEND
+	select MBEDTLS_TLS_LIBRARY
+	select MBEDTLS_AES_C
+	select MBEDTLS_RSA_C
+	select MBEDTLS_ECJPAKE_C
+	select MBEDTLS_ECP_C
+	select MBEDTLS_ENABLE_HEAP
+	select NRF_SECURITY_ADVANCED
+	help
+	  Enables nrf_security module for use by OpenThread. This allows
+	  OpenThread to make use of hardware accelerated cryptography functions
+	  if available as well as fast oberon backend for software encryption.
+
+config OPENTHREAD_MBEDTLS_TARGET
+	string "mbedtls target name"
+	default "mbedtls_common" if OPENTHREAD_NRF_SECURITY
+	help
+	  This option allows to specify custom mbedtls CMake target name for
+	  OpenThread. This is sometimes needed to utilize hardware cryptography
+	  acceleration.
```

Comment:
Why is this being defined here, as it is also available together with openthread sources upstream:
https://github.com/nrfconnect/sdk-zephyr/blob/master/subsys/net/l2/openthread/Kconfig#L52

Secondly, there seems to be missing a depends, cause `OPENTHREAD_MBEDTLS_TARGET` is only used here:
https://github.com/zephyrproject-rtos/openthread/blob/zephyr/zephyr/CMakeLists.txt#L13
and so is only used when `CONFIG_OPENTHREAD_SOURCES=y`, see: https://github.com/zephyrproject-rtos/openthread/blob/zephyr/zephyr/CMakeLists.txt#L1

Seems to me this setting is only relevant when compiling from sources, where you can also specify the value:
![image](https://user-images.githubusercontent.com/9802583/90251731-8c376280-de3e-11ea-9acc-3f0125eddc58.png)

what I think you want, is to just have another default value, but no prompt, right ?

Reply:
Agree - there is no need for additional OPENTHREAD_MBEDTLS_TARGET definition but for another default value.

My proposition is to change [this line](https://github.com/nrfconnect/sdk-zephyr/blob/master/subsys/net/l2/openthread/Kconfig#L52) to the following:

config OPENTHREAD_MBEDTLS_TARGET
	string "mbedtls target name"
        default "mbedtls_common" if !OPENTHREAD_MBEDTLS
        default "mbedTLS" 
	depends on OPENTHREAD_SOURCES
	help
	  This option allows to specify custom mbedtls CMake target name for
	  openthread. This is sometimes needed to utilize hardware cryptography
	  acceleration.

This would also add dependency on OPENTHREAD_SOURCES.

Another possibility is to add `CONFIG_OPENTHREAD_MBEDTLS_TARGET="mbedtls_common"` at the end of [config file](https://github.com/nrfconnect/sdk-nrf/blob/96ce0c458acf5c4d0b39d0a45702814e67951236/samples/openthread/cli/boards/nrf52840dk_nrf52840.conf#L8).

@tejlmand, what do you think?

Reply:
but that would require an upstream change in Zephyr.
If you can post a PR upstream, I guess that would be fine, but i'm not sure upstream is interested in this line:
```
 default "mbedtls_common" if !OPENTHREAD_MBEDTLS
```
That is only something we need downstream, right ?
(disclaimer, i'm not familiar with Openthread code upstream).

But for NCS, you have the option of just specifying another default value in Kconfig, by simply remove the prompt:
```suggestion
config OPENTHREAD_MBEDTLS_TARGET
	default "mbedtls_common" if OPENTHREAD_NRF_SECURITY
```
and it will give you another default value when `OPENTHREAD_NRF_SECURITY=y`:
![image](https://user-images.githubusercontent.com/9802583/90511905-881c8500-e15d-11ea-9257-c4720cd2e279.png)

and in NCS , users will see:
![image](https://user-images.githubusercontent.com/9802583/90512002-a97d7100-e15d-11ea-8afb-8741b7f13cfd.png)

without patching Zephyr.

> Another possibility is to add CONFIG_OPENTHREAD_MBEDTLS_TARGET="mbedtls_common" at the end of config file.

I don't think that is needed here.
You want this default everywhere `OPENTHREAD_NRF_SECURITY` is used, right ?

Reply:
@tejlmand Thank you for your comment.

Yes, if CONFIG_OPENTHREAD_MBEDTLS_TARGET is used when CONFIG_OPENTHREAD_NRF_SECURITY is set, the value should always be 'mbedtls_common'.

The solution you provided looks good to me, I was not aware of the possibility of doing that. Will push commit with that change.

Apart of that, I will introduce a PR to upstream that fixes the dependency issue.